source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+communication between pods requires additional configuration. If you want your
+pods to be able to communicate with each other you must choose and install a
+`pod network add-on`_. Otherwise, this section can be skipped. 
 
-   or:
+This is important to decide early because some pod network add-ons require
+configuration during cluster initialization. Check whether or not your add-on
+requires special flags when you initialize the master control plane.
 
-   b) Permanently:
+If your chosen network add-on requires appending to the :command:`kubeadm
+init` command, make note of it before continuing. 
 
-      Mask the swap partition:
+For example, if you choose the *flannel* pod network add-on, then in later
+steps you must add the following to the :command:`kubeadm init` command:
 
-      .. code-block:: bash
+.. code-block:: console
+
+   --pod-network-cidr 10.244.0.0/16
 
-         sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
-         sudo swapoff -a
 
-      .. note::
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
+Choose a container runtime
+==========================
 
-#. Switch to root to modify the `hosts` file:
+Container runtimes are the underlying fabric that pod workloads execute inside
+of. Different container runtimes offer different balances between features,
+performance, and security. 
+
+Kubernetes allows integration various container runtimes via a container
+runtime interface (CRI). |CL| supports Kubernetes with the various runtimes
+below:
+
+* `Docker (Dockershim)`_ with or without `Kata Containers`_.
+* `containerd`_ with or without `Kata Containers`_.
+* `CRI+O`_ with or without `Kata Containers`_.
+
+
+The container runtime that you choose will dictate the steps necessary to
+initialize the master cluster with :command:`kubeadm init`.
+
+
+Docker (Dockershim)
+-------------------
+
+The default built-in runtime provided by Kubernetes is using the system Docker
+via Dockershim and as a result is one of the simplest to use. 
+
+One limitation of using Dockershim is that all pods on the Kubernetes node
+will inherit and use the default runtime that Docker is set to use. To be able
+to specify a container runtime per-Kerbernetes service, use CRI+O or
+containerd. 
+
+
+#. Make sure Docker is installed:
+
+   .. code:: bash
+
+      sudo swupd bundle-add containers-basic
+
+#. Start the Docker service and enable it to start automatically at boot:
+
+   .. code::
+
+      sudo systemctl enable --now docker.service
+
+
+#. Run the kubeadm command to initialize the master node:
+
+   .. important:: 
+
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code:: bash
+
+      sudo kubeadm init \
```

Comment:
I think showing two examples both with and without would be more helpful.

Reply:
Do you mean one example without flannel and one with? If so, that is one of the things I wanted to simplify. Examples trying to cover all the different cases is hard to follow.

Reply:
I meant show one example where you init and don't have to add any extra parameters and then another example where you do add the extra parameters.  When you reference that pod network add-on link you can see how the extra parameters fit into the overall init command.   I'm not sure how you could fit this into the doc but seeing 5 or 6 different init commands at one time showing how the different CRI runtimes are started and how different network plugins are enabled are always super helpful for me.  It allows my brain to easily see what is the Kubernetes portion of the init, what is the crio/docker/containerd portion of the init, and what are the networking addin portions of the init and how they can vary.